### DISCUSSION TITLE
JSON or XML Feed for Calendar

### DISCUSSION THREAD

**Discussion by laurabennett** [solspace/craft-calendar#240] [Questions & Answers] [answered]
Another developer is creating an app for our client and looking to pull in calendar events through an online feed. Hoping this is possible to do either directly from within Calendar or by adding another plugin. I think there are some plugins that work with Craft entries but not sure  how that works with Calendar.
Thanks!

**Comment by kjmartens** [comment]
Sorry for the delay...

The closest you could get to this (pulling external events into Calendar, if I am understanding correctly) is to make use of the [FeedMe](https://plugins.craftcms.com/feed-me) plugin. You would need to set up your external site to expose event data through a URL, and then could have a cron job set to trigger that _FeedMe_ profile. 🙂 

https://docs.solspace.com/craft/calendar/v4/overview/importing/

If you're referring to exporting Calendar event to import into another app or site, you could make use of the ICS Subscriptions feature:
https://docs.solspace.com/craft/calendar/v4/overview/exporting/

**Reply by laurabennett** [reply]
I've used FeedMe to pull in events and that's not an issue. What we need to provide an ongoing feed not a one time export. Isn't ICS fairly specific to Outlook? We are looking for XML or JSON.
Thanks

**Reply by kjmartens** [reply, marked as answer]
Sorry, I somewhat misunderstood your request then.

If you're looking to generate an XML or JSON feed of Calendar events, you could do this in Twig and build it manually. The Calendar demo templates does this already (`/resources/event.json.html`) to make Calendar work with the [FullCalendar](https://fullcalendar.io/) library.

This uses a bit of a shortcut approach, but you could go about it differently: 🙂 

``` twig
{% set events = craft.calendar.events({
    rangeStart: craft.app.request.post("rangeStart"),
    rangeEnd: craft.app.request.post("rangeEnd"),
    limit: 500,
}) %}
{{ events.all|json_encode|raw }}
```

**Reply by laurabennett** [reply]
I'll give this a try thanks!!
